io-sim/test/Test/IOSim.hs

```diff
@@ -180,17 +181,17 @@ instance Arbitrary TestMicro where
       genN :: Int -> [Micro] -> Gen [Micro]
       genN 0 rs = return rs
       genN n [] = do
-        r <- genPositiveMicro
+        r <- genMicro
         genN (n - 1) [r]
       genN n rs = do
         r <- frequency
           [ (2, elements rs)
-          , (1, genPositiveMicro)
+          , (1, genMicro)
           ]
         genN (n - 1) (r : rs)
 
-      genPositiveMicro :: Gen Micro
-      genPositiveMicro = MkFixed . getPositive <$> arbitrary
+      genMicro :: Gen Micro
+      genMicro = MkFixed <$> arbitrary
```

Comment:
Ahhh. I was misreading before. Now we allow negative timeouts so we can test them. Yes of course :-)